build.rs

```diff
@@ -1,27 +1,41 @@
+extern crate bindgen;
 extern crate cmake;
 
 fn main() {
-    let mut cfg = cmake::Config::new("openvr");
+    let mut config = cmake::Config::new("openvr");
 
-    // Work around broken cmake build
-    #[cfg(windows)]
-    cfg.cxxflag("/DWIN32");
+    if cfg!(target_os = "macos") {
+        config.define("BUILD_UNIVERSAL", "OFF");
+    } else if cfg!(target_os = "windows") {
+        config.cxxflag("/DWIN32");
+    }
 
-    #[cfg(target_os="macos")]
-    cfg.define("BUILD_UNIVERSAL", "OFF");
-
-    let dst = cfg.build();
+    let dst = config.build();
     println!("cargo:rustc-link-search=native={}/lib", dst.display());
 
-    #[cfg(all(windows, target_pointer_width = "64"))]
-    println!("cargo:rustc-link-lib=static=openvr_api64");
-
-    #[cfg(not(all(windows, target_pointer_width = "64")))]
-    println!("cargo:rustc-link-lib=static=openvr_api");
-
-    #[cfg(target_os="linux")]
-    println!("cargo:rustc-link-lib=stdc++");
-
-    #[cfg(target_os="macos")]
-    println!("cargo:rustc-link-lib=c++");
+    if cfg!(target_os = "linux") {
+        println!("cargo:rustc-link-lib=stdc++");
+        println!("cargo:rustc-link-lib=static=openvr_api");
+    } else if cfg!(target_os = "macos") {
+        println!("cargo:rustc-link-lib=c++");
+        println!("cargo:rustc-link-lib=static=openvr_api");
+    } else if cfg!(target_os = "windows") {
+        println!("cargo:rustc-link-lib=shell32");
+        if cfg!(target_pointer_width = "64") {
+            println!("cargo:rustc-link-lib=static=openvr_api64");
+        } else {
+            println!("cargo:rustc-link-lib=static=openvr_api");
+        }
+    }
```

Comment:
I think this could be simplified by pulling out the `openvr_api` print to a separate if chain with cases for `cfg!(target_os = "windows") && cfg!(target_pointer_width = "64")`. That said, isn't this (both old and new) incorrectly operating in terms of the build script's target, i.e. the build host, rather than the build target?